hazelcast/src/main/java/com/hazelcast/client/impl/protocol/task/map/DefaultMapAggregateMessageTask.java

```diff
@@ -76,6 +80,19 @@ protected Object reduce(Collection<AggregationResult> results) {
                 combinedResult.onCombineFinished();
             }
         }
-        return combinedResult != null ? combinedResult.getAggregator().aggregate() : null;
+
+        if (combinedResult == null) {
+            return null;
+        }
+
+        Object result = combinedResult.getAggregator().aggregate();
+        if (result instanceof CanonicalizingHashSet && endpoint.getClientVersion() < MIXED_TYPES_VERSION) {
+            // XXX: Older Java clients are expecting a HashSet instance
+            // serialized using standard java.io.Serializable facilities.
+            return new HashSet((CanonicalizingHashSet) result);
+        }
+
+        return result;
```

Comment:
Since this is used by only java client the change seems ok.  We are using IdentifiedDAtaSerializable as long as they are configured on both on the member and on the client. Since we are doing a version check, all clients will have the IDS configured correctly.
Version check was a question for me, but since we did the same thing on `ClientHeartbeatMonitor` I think we are ok with it. 

Can you add compatibility tests to verify this part is working correctly?

Reply:
@sancar added the compatibility test in the EE counterpart: https://github.com/hazelcast/hazelcast-enterprise/pull/2836/commits/2394f408198241121865305cf51e1a7f45c8ceeb, client compatibility tests are possible in EE repo only.